robottelo/test.py

```diff
@@ -0,0 +1,323 @@
+    _default_notraises_value_handler = None
+
+    @classmethod
+    def setUpClass(cls):  # noqa
+        super().setUpClass()
+        if not settings.configured:
+            settings.configure()
+        cls.logger = logger
+        cls.logger.info(f'Started setUpClass: {cls.__module__}/{cls.__name__}')
+        # NOTE: longMessage defaults to True in Python 3.1 and above
+        cls.longMessage = True
+        cls.foreman_user = settings.server.admin_username
+        cls.foreman_password = settings.server.admin_password
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.logger.info(f'Started tearDownClass: {cls.__module__}/{cls.__name__}')
+
+    @classmethod
+    def upload_manifest(cls, org_id, manifest, interface=None, timeout=None):
+        """Upload manifest locked using the default TestCase manifest if
+        interface not specified.
+
+        Usage::
+
+            manifest = manifests.clone()
+            self.upload_manifest(org_id, manifest)
+
+            # or if you want to specify explicitly an interface
+            manifest = manifests.clone()
+            self.upload_manifest(org_id, manifest, interface=INTERFACE_CLI)
+
+            # in one line
+            result = self.upload_manifest(org_id, manifests.clone())
+        """
+        if interface is None:
+            interface = cls._default_interface
+        if timeout is None:
+            # upload the manifest with default ssh client command timeout.
+            timeout = settings.ssh_client.command_timeout
+        return manifests.upload_manifest_locked(
+            org_id, manifest, interface=interface, timeout=timeout
+        )
+
+    def assertNotRaises(
+        self,
+        expected_exception,
+        callableObj=None,
+        expected_value=None,
+        value_handler=None,
+        *args,
+        **kwargs,
+    ):
+        """Fail if an exception of class expected_exception is raised by
+        callableObj when invoked with specified positional and keyword
+        arguments. If a different type of exception is raised, it will not be
+        caught, and the test case will be deemed to have suffered an error,
+        exactly as for an unexpected exception.
+
+        If called with callableObj omitted or None, will return a context
+        object used like this::
+
+                with self.assertNotRaises(SomeException):
+                    do_something()
+
+        The context manager keeps a reference to the exception as the
+        'exception' attribute. This allows you to inspect the exception after
+        the assertion::
+
+               with self.assertNotRaises(SomeException) as cm:
+                   do_something()
+               the_exception = cm.exception
+               self.assertEqual(the_exception.error_code, 1)
+
+        In addition, optional 'http_status_code' or 'cli_return_code' arg may
+        be passed. This allows to specify exact HTTP status code or CLI return
```

Comment:
```suggestion
        be passed. This enables specifying exact HTTP status code or CLI return
```

Reply:
Ha!! This entire module should not be here. This branch was created before the PRs went in removing it, and since I had changes to it the rebase resolved to keep the entire thing.